subworkflows/local/bam_variant_calling_somatic_mutect2_ms/main.nf

```diff
@@ -0,0 +1,324 @@
+include { GATK4_GETPILEUPSUMMARIES        as GETPILEUPSUMMARIES_TUMOR      } from '../../../modules/nf-core/gatk4/getpileupsummaries/main'
+include { GATK4_CALCULATECONTAMINATION    as CALCULATECONTAMINATION        } from '../../../modules/nf-core/gatk4/calculatecontamination/main'
+include { GATK4_FILTERMUTECTCALLS         as FILTERMUTECTCALLS             } from '../../../modules/nf-core/gatk4/filtermutectcalls/main'
+
+
+workflow BAM_VARIANT_CALLING_SOMATIC_MUTECT2_MS {
+    take:
+    input                     // channel: [ meta, [ input ], [ input_index ] ]
+    fasta                     // channel: /path/to/reference/fasta
+    fai                       // channel: /path/to/reference/fasta/index
+    dict                      // channel: /path/to/reference/fasta/dictionary
+    germline_resource         // channel: /path/to/germline/resource
+    germline_resource_tbi     // channel: /path/to/germline/index
+    panel_of_normals          // channel: /path/to/panel/of/normals
+    panel_of_normals_tbi      // channel: /path/to/panel/of/normals/index
+    intervals                 // channel: [mandatory] [ intervals, num_intervals ] or [ [], 0 ] if no intervals
+
+    main:
+    ch_versions = Channel.empty()
+    germline_resource_pileup     = germline_resource_tbi ? germline_resource : Channel.empty()
+    germline_resource_pileup_tbi = germline_resource_tbi ?: Channel.empty()
+
+
+    // Combine input and intervals for spread and gather strategy
+    input_intervals = input.combine(intervals)
+        // Move num_intervals to meta map and reorganize channel for MUTECT2_PAIRED module
+        .map{ meta, n_cram, n_crai, t_cram, t_crai, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], n_cram, n_crai, t_cram, t_crai, intervals ] }
+
+    // Prepare multi sample Mutect2 variant calling channels
+    // Separate normal cram files
+    ch_normal_cram = input_intervals.map{meta, n_cram, n_crai, t_cram, t_crai, intervals -> [[id: meta.patient, patient: meta.patient, normal_id: meta.normal_id, sex: meta.sex, num_intervals: meta.num_intervals], n_cram, n_crai]}.unique().groupTuple()
+    // If there are multiple normal cram files, use the first one only
+    ch_normal_cram_first = ch_normal_cram.map{it -> [it[0], it[1][0], it[2][0]]}
+    // Separate tumor cram files
+    ch_tumor_cram = input_intervals.map{meta, n_cram, n_crai, t_cram, t_crai, intervals -> [[id: meta.patient, patient: meta.patient, normal_id: meta.normal_id, sex: meta.sex, num_intervals: meta.num_intervals], t_cram, t_crai]}.unique()
+    // Merge normal and tumor samples by patient
+    ch_tn_cram = ch_normal_cram_first.mix(ch_tumor_cram).groupTuple()
+    // Add intervals back
+    ch_pt_intervals = input_intervals.map{meta, n_cram, n_crai, t_cram, t_crai, intervals -> [[id: meta.patient, patient: meta.patient, normal_id: meta.normal_id, sex: meta.sex, num_intervals: meta.num_intervals], intervals]}.unique().groupTuple()
```

Comment:
the meta mapping can be done with addition/subMap, which is safer then manually setting all keys, something we learned recently and have updated in all other places